src/pugixml.hpp

```diff
@@ -122,6 +122,26 @@
 #	endif
 #endif
 
+// Detect if C++ standard supports 'string_view' (2017 or higher)
+#ifndef PUGI_STRING_VIEW_SUPPORTED
+#	if defined(__cpp_lib_string_view) || __cplusplus >= 201703L || defined(_MSVC_LANG) && _MSVC_LANG >= 201703L
```

Comment:
```suggestion
#	if defined(__cpp_lib_string_view) || __cplusplus >= 201703L || (defined(_MSVC_LANG) && _MSVC_LANG >= 201703L)
```

Reply:
Also, out of curiosity, what is the distinction between _cpp_lib & __cplusplus checks here? Do we need both and if so, for what purpose?

Reply:
In principle the [feature testing macros](https://en.cppreference.com/w/cpp/feature_test) are a standardized way of  directly testing feature support. If they are available then it is less hassle than testing `__cplusplus` (thanks to `_MSVC_LANG`) and they are more expressive.
Sadly, they are only guaranteed with C++20 although the big three provide them earlier. (if I researched correctly)

 So `__cpp_lib_string_view` is neither required nor sufficient.

I wrote it like that because [someone on Microsoft's STL implementation](https://github.com/microsoft/STL/issues/2503#issuecomment-1021063613) proposed this.
And I can see one advantage. If your compiler does not support the entirety of C++17 but already supports `string_view` and defines the macro accordingly then support would be enabled.

Reply:
So seeing that the options are to either remove the check on `__cpp_lib_string_view` or to also add
```
#if defined(__has_include) && __has_include(<version>)
#   include <version>
#endif
```
gives a pretty good idea which option you'll prefer.